### ISSUE TITLE
[Bug]: cast decimal128 to decimal64

### ISSUE THREAD

**Issue by ouyuanning** [matrixorigin/matrixone#3806]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

[SQL STATEMENT]: select cast(9223372.036854775808 as decimal)+1;
[EXPECT RESULT]:
cast(9223372.036854775808 as decimal)+1
9223373
[ACTUAL RESULT]:
Operator 'cast' with parameters [DECIMAL128 DECIMAL64] will be implemented in future version.

**Comment by ouyuanning** [comment]
when fix #3785  we get #3806

**Comment by ouyuanning** [comment]
fixed it in #3785 
thanks @broccoliSpicy
